src/main/java/net/starschema/clouddb/jdbc/BQStatementRoot.java

```diff
@@ -333,21 +335,40 @@ public ResultSet executeQuery(String querySql, boolean unlimitedBillingBytes)
               this.connection.getUseQueryCache());
       this.mostRecentJobReference.set(qr.getJobReference());
 
+      referencedJob =
+          this.connection
+              .getBigquery()
+              .jobs()
+              .get(projectId, qr.getJobReference().getJobId())
+              .execute();
+
       if (defaultValueIfNull(qr.getJobComplete(), false)) {
         List<TableRow> rows = defaultValueIfNull(qr.getRows(), new ArrayList<TableRow>());
         if (BigInteger.valueOf(rows.size()).equals(qr.getTotalRows())) {
           TableSchema schema = defaultValueIfNull(qr.getSchema(), new TableSchema());
+          BiEngineStatistics biEngineStatistics = null;
```

Comment:
Similar to previous comment, you can move this declaration inside the `if (referencedJob != null)` check so that its scope is more clearly defined.